File: 백준/Silver/1244. 스위치 켜고 끄기/스위치 켜고 끄기.cc
#include <iostream>
using namespace std;

int N;
int switches[101];

int main(void) {
    cin >> N;
    for (int i = 1; i <= N; i++) cin >> switches[i];
    int students;
    cin >> students;
    for (int _ = 0; _ < students; _++) {
        int gender, num;
        cin >> gender >> num;
        if (gender == 1) {
            for (int j = 1; j <= N; j++)
                if (j % num == 0) switches[j] = 1 - switches[j];
        } else {
            switches[num] = 1 - switches[num];
            int idx = 1;
            while (num + idx <= N && num - idx >= 1 && switches[num+idx] == switches[num-idx]) {
                switches[num + idx] = 1 - switches[num + idx];
                switches[num - idx] = 1 - switches[num - idx];
                idx++;
            }
        }
    }
    for (int i = 1; i <= N; i++) {
        printf("%d ", switches[i]);
        if (i % 20 == 0) printf("\n");
    }
    printf("\n");
}

File: 백준/Silver/1244. 스위치 켜고 끄기/스위치 켜고 끄기.py
def tiktok(gender, number):
    # 남학생
    if gender == 1:
        for i in range(0, N):
            if (i + 1) % number == 0:
                switches[i] = 1 - switches[i]
    # 여학생
    else:
        number -= 1
        switches[number] = 1 - switches[number]
        idx = 1
        while number + idx < N and number - idx >= 0\
            and switches[number + idx] == switches[number - idx]:
            switches[number + idx] = 1 - switches[number + idx]
            switches[number - idx] = 1 - switches[number - idx]
            idx += 1


N = int(input())
switches = list(map(int, input().split()))
query_num = int(input())
queries = [tuple(map(int, input().split())) for _ in range(query_num)]
for query in queries:
    tiktok(*query)

stk = 0
for switch in switches:
    print(switch, end=' ')
    stk += 1
    if stk == 20:
        print()
        stk = 0
